cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -44,6 +46,73 @@ def _create_secret(secret: Dict[Text, Text]):
         logging.info(f"Secret '{secret['secret_name']}' created.")
 
 
+def create_job(
+    pod_name: Text,
+    image: Text,
+    tag: Text,
+    env_variables: List[Dict[Text, Text]],
+    secrets: List[Dict[Text, Text]],
+    command: List[Text],
+    args: List[Text],
+    dry_run: bool,
+    node_selector: Optional[Dict[Text, Text]] = None,
+    labels: Optional[Dict[Text, Text]] = None,
+    extra_arg: Optional[Text] = None,
+) -> Text:
```

Comment:
we can simplify the signature by passing in the output of `_make_job_spec` instead of 95% of the arguments.

also seems logical to break this long function into smaller ones - 

delete_jobs, wait, create